towhee/engine/operator_loader.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import importlib.util
+from pathlib import Path
+from typing import Any, Dict
+
+from towhee.operator.base import OperatorBase
+
+
+class OperatorLoader:
+    """Wrapper class used to load operators from either local cache or a remote
+    location.
+
+    Args:
+        cache_path: (`str`)
+            Local cache path to use. If not specified, it will default to
+            `$HOME/.towhee/cache`.
+    """
+
+    def __init__(self, cache_path: str = None):
+        if not cache_path:
+            self._cache_path = Path.home() / '.towhee/cache'
+        else:
+            self._cache_path = Path(cache_path)
+        self._cache_path = Path(cache_path)
+
+    def load_operator(self, function: str, args: Dict[str, Any]) -> OperatorBase:
+        """Attempts to load an operator from cache. If it does not exist, looks up the
+        operator in a remote location and downloads it to cache instead. By standard
+        convention, the operator must be called `Operator` and all associated data must
+        be contained within a single directory.
+
+        Args:
+            function: (`str`)
+                Origin and method/class name of the operator. Used to look up the proper
+                operator in cache.
+        """
+
+        # Lookup the path for the operator in local cache. Example directory structure:
+        #  /home/user/.towhee/cache/organization-name/operator-name
+        #   |_ /home/user/.towhee/cache/organization-name/operator-name/operator.py
+        #   |_ /home/user/.towhee/cache/organization-name/operator-name/resnet.torch
+        #   |_ /home/user/.towhee/cache/organization-name/operator-name/config.json
+        paths = list((self._cache_path / function).glob('[!_]*.py'))
+        if not paths:
+            raise TypeError('Operator definition not found')
+        if len(paths) > 1:
```

Comment:
An operator can have many *.py files, we just use operator.py to create an operator object.

Reply:
I initially had it just named as `operator.py`, but during testing phase it conflicted with the `operator` standard library. Perhaps just `definition.py` or `operator_def.py`?